### ISSUE TITLE
Unnecessary "confirm query execution" dialog in some queries

### ISSUE THREAD

**Issue by ewoerner** [dbeaver/dbeaver#19933]
### Description

The following query triggers a "Confirm query execution" dialog, which is unnecessary since it's a read-only query (simplified):
```
select bool_or(true = true); -- triggers confirmation dialog
```
The following related queries work fine, i.e. they do not trigger the confirmation dialog:
```
select true = true; -- ok
select bool_or(true); -- ok
```

### DBeaver Version

Community Edition 23.0.1

### Operating System

Windows 11

### Database and driver

PostgreSQL 14.5
Driver Version 42.5.0

**Comment by ShadelessFox** [comment]
Hello @ewoerner,

DBeaver doesn't know whether a particular query might or not have side effects. Therefore, it always assumes there can be one.

I think the current behavior should be kept.

**Comment by ewoerner** [comment]
@ShadelessFox that is not correct, DBeaver seems to have some logic (`SQLSemanticProcessor.isSelectQuery`) that decides whether a query might have side effects. See the two related queries in my original comment which do not trigger the confirm query execution dialog.

**Comment by ShadelessFox** [comment]
Perhaps the existing semantics of the `SQLSemanticProcessor#isSelectQuery` function are invalid, actually.

In your example, the `bool_or` functions can have side effects, but DBeaver doesn't complain about it for some reason.

There's room for improvement. I'm sorry for sweeping the issue right away without a proper investigation.

**Comment by arhayka** [comment]
Could you please try to update DBeaver and check this in the last version?

**Comment by ewoerner** [comment]
Still reproducible in 23.0.4

**Comment by ShadelessFox** [comment]
I'm unable to reproduce the issue. Can you confirm if it's reproducible against 23.0.5 or wait until 23.1.0 is released on Monday?

**Comment by ewoerner** [comment]
I can reproduce using 23.1.0 (EA)

**Comment by ShadelessFox** [comment]
Thanks.

**Comment by uslss** [comment]
https://github.com/dbeaver/dbeaver/assets/31996417/81fb3c0b-8817-4c2b-9de6-7d7457e7c798

Reproducible only if "Confirm SQL Execution" is enabled for connection type

**Comment by arhayka** [comment]
@ewoerner Thank you for your request.
We need to enhance our parser to improve this behavior.
